stingray/tests/test_base.py

```diff
@@ -84,6 +69,197 @@ def test_instantiate_without_main_array_attr(self):
         with pytest.raises(RuntimeError):
             BadStingrayObj(self.arr)
 
+    def test_equality(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        assert ts1 == ts2
+
+    def test_different_array_attributes(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        # Add a meta attribute only to ts1. This will fail
+        ts1.blah = 2
+        assert ts1 != ts2
+
+        # Add a non-scalar meta attribute, but with the same name, to ts2.
+        ts2.blah = [2]
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah, ts2.blah
+        assert ts1 == ts2
+
+        # Add a non-scalar meta attribute to both, just slightly different
+        ts1.blah = [2]
+        ts2.blah = [3]
+        ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah, ts2.blah
+        assert ts1 == ts2
+
+        # Add a meta attribute only to ts2. This will also fail
+        ts2.blah = 3
+        assert ts1 != ts2
+
+    def test_different_meta_attributes(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        # Add an array attribute to ts1. This will fail
+        ts1.blah = ts1.guefus * 2
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah
+        assert ts1 == ts2
+        # Add an array attribute to ts2. This will fail
+        ts2.blah = ts1.guefus * 2
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts2.blah
+        assert ts1 == ts2
+
+    def test_apply_mask(self):
+        ts = copy.deepcopy(self.sting_obj)
+        newts0 = ts.apply_mask([True, True, False], inplace=False)
+        newts1 = ts.apply_mask([True, True, False], inplace=True)
+        assert newts0.parafritus == "bonus!"
+        assert newts1.parafritus == "bonus!"
+        for obj in [newts1, newts0]:
+            assert obj.parafritus == "bonus!"
+            assert np.array_equal(obj.guefus, [4, 5])
+            assert np.array_equal(obj.panesapa, ts.panesapa)
+            assert np.array_equal(obj.pardulas, [3.0 + 1.0j, 2.0j])
+            assert np.array_equal(obj.sebadas, [[0, 1], [2, 3]])
+        assert ts is newts1
+        assert ts is not newts0
+
+    @pytest.mark.parametrize("inplace", [True, False])
+    def test_partial_apply_mask(self, inplace):
+        ts = copy.deepcopy(self.sting_obj)
+        obj = ts.apply_mask([True, True, False], inplace=inplace, filtered_attrs=["pardulas"])
```

Comment:
This test is still on the base class `StingrayObject`, and in this particular example `guefus` is the main array attribute (like `time` in `Lightcurve` or `StingrayTimeseries`, or `freq` in `Crossspectrum`). So it will get filtered by default. `filtered_attrs` only lists the attributes besides the main array attribute